src/transformers/visitors/checks/interactsWithSecretVisitor.ts

```diff
@@ -34,6 +34,30 @@ const markSubtreeInteractsWithPublic = (thisPath: any, thisState: any) => {
     indicator.addPublicInteractingPath(thisState.publicPath);
 };
 
+const markIndicatorSubtreeInteractsWithSecret = (thisPath: any, thisState: any) => {
+  const { node, scope } = thisPath;
+  if (!['Identifier', 'VariableDeclarationStatement'].includes(node.nodeType))
+    return;
+  const indicator = scope.getReferencedIndicator(node, true);
+  // we don't want to add itself as an interacted with path
+  if (indicator && thisState.secretPath.node.id !== node.id)
+    indicator.addSecretInteractingPath(thisState.secretPath);
+};
+
+const interactwithPublicSecret = (thisPath: any, thisState: any) => {
```

Comment:
This function is not used anywhere, do we need this?

Reply:
This is used on line 79.

Reply:
But are we using interactwithPublicSecret indicator anywhere ?

Reply:
No, we use this function to set interactsWithSecret not interactsWithPublicSecret. We could change the name of this? Maybe to something like inferInteractsWithSecret?